src/server/redis_connection.cc

```diff
@@ -396,6 +413,13 @@ void Connection::ExecuteCommands(std::deque<CommandTokens> *to_process_cmds) {
     auto cmd_name = attributes->name;
     auto cmd_flags = attributes->GenerateFlags(cmd_tokens);
 
+    // Pause the processing of only the write commands, and push them back
+    // to a list that adds them back to the queue to process them when we unpause
+    if (srv_->GetCommandPauseType() == kPauseWrite && (cmd_flags & kCmdWrite)) {
```

Comment:
I didn't check the Redis codebase, but it looks like the entire connection will be suspended like `PAUSE ALL` once found the write command from the documentation: https://redis.io/docs/latest/commands/client-pause/

> WRITE: Clients are only blocked if they attempt to execute a write command.

And from the current implementation, it will let read commands go through and it might cause unmatched responses in client side.

Reply:
[Looking at the Redis codebase](https://github.com/redis/redis/blob/3264deb24e3b01ad7b2777bc68b9d91f0f41aa3b/src/server.c#L4186), it seems to me that the server allows read commands to pass through when we are in `PAUSE WRITE` mode, and `PAUSE ALL` is when we defer all. Let me know if I got it right.
Also slaves should be kept exempt from this as per Redis, so I'll be working on that.

Reply:
@furkan-bilgin Yes, what I mean is that the client should be pending once it occurred the WRITE command. However, the current implementation will let READ commands go through.

Reply:
Alright, now I got it. Pushed the new commit.